src/Component/AbstractComponent.php

```diff
@@ -270,12 +270,22 @@ final public function toManagementApiData () : array
 
 		$definition = $this->configureComponent();
 		$previewField = null;
+		$previewImageField = null;
 
 		foreach ($this->getFields()->getRootFields() as $key => $field)
 		{
 			if ($field->isStoryblokPreviewField())
 			{
 				$previewField = $key;
+			}
+
+			if ($field->isStoryblokImagePreviewField())
+			{
+				$previewImageField = $key;
+			}
+
+			if (null !== $previewField && null !== $previewImageField)
+			{
 				break;
```

Comment:
It now can't happen that we have two different preview field vs asset preview, right? Is that something that can happen in theory? Or can it only ever be one?